drivers/hmis/app/jobs/hmis/merge_clients_job.rb

```diff
@@ -223,6 +224,27 @@ def merge_mci_ids
         update_all(source_id: client_to_retain.id)
     end
 
+    # Note: WarehouseChangesJob process kicks off MergeClientsJob for clients that
+    # share the same MCI Unique ID, so that is the most likely scenario. However,
+    # it's also possible to perform a manual merge in HMIS for two clients that
+    # may or may not share MCI Unique ID values.
+    def merge_mci_unique_ids
+      # If retained client has an MCI Unique ID, no action is needed.
+      # Max 1 MCI Unique ID is permitted per client, so if any of
+      # the merged clients have differing MCI Unique IDs, they will be destroyed.
+      # (This could happen in the case of a manual merge).
+      return if client_to_retain.ac_hmis_mci_unique_id.present?
+
+      # If retained client does not have an MCI Unique ID, try to find one to keep from the merged clients
+      mci_unique_id_to_keep = HmisExternalApis::ExternalId.mci_unique_ids.
+        where(source: clients_needing_reference_updates).
+        max_by(&:updated_at)
+      return unless mci_unique_id_to_keep
+
+      # Re-assign this MCI Unique ID to the retained client
+      mci_unique_id_to_keep.update!(source: client_to_retain)
```

Comment:
Is there any argument here for keeping the old ExternalID and creating a new one for the new client (with the old MCI ID) or is there no reason to retain that history?

Reply:
I don't think that's necessary- this here matches the behavior for MCI ID so I think better to keep them consistent